commands/payment_channel_daemon_test.go

```diff
@@ -1,547 +1,620 @@
 	"github.com/filecoin-project/go-filecoin/address"
-	"github.com/filecoin-project/go-filecoin/fixtures"
 	th "github.com/filecoin-project/go-filecoin/testhelpers"
+	"github.com/filecoin-project/go-filecoin/tools/fast"
+	"github.com/filecoin-project/go-filecoin/tools/fast/series"
+	localplugin "github.com/filecoin-project/go-filecoin/tools/iptb-plugins/filecoin/local"
 	"github.com/filecoin-project/go-filecoin/types"
 )
 
-func TestPaymentChannelCreateSuccess(t *testing.T) {
-	t.Parallel()
-	assert := assert.New(t)
-
-	d := makeTestDaemonWithMinerAndStart(t)
-	defer d.ShutdownSuccess()
-
-	args := []string{"paych", "create"}
-	args = append(args, "--from", fixtures.TestAddresses[0], "--gas-price", "0", "--gas-limit", "300")
-	args = append(args, fixtures.TestAddresses[1], "10000", "20")
+func init() {
+	_, _ = os.Getwd()
 
-	paymentChannelCmd := d.RunSuccess(args...)
-	messageCid, err := cid.Parse(strings.Trim(paymentChannelCmd.ReadStdout(), "\n"))
-	require.NoError(t, err)
-
-	var wg sync.WaitGroup
+	series.GlobalSleepDelay = time.Second * 5
+}
 
-	wg.Add(1)
-	go func() {
-		wait := d.RunSuccess("message", "wait",
-			"--return",
-			"--message=false",
-			"--receipt=false",
-			messageCid.String(),
-		)
-		_, ok := types.NewChannelIDFromString(strings.Trim(wait.ReadStdout(), "\n"), 10)
-		assert.True(ok)
-		wg.Done()
-	}()
+func VoucherFromString(data string) (paymentbroker.PaymentVoucher, error) {
+	_, cborVoucher, err := multibase.Decode(data)
+	if err != nil {
+		return paymentbroker.PaymentVoucher{}, err
+	}
 
-	d.RunSuccess("mining once")
+	var voucher paymentbroker.PaymentVoucher
+	if err = cbor.DecodeInto(cborVoucher, &voucher); err != nil {
+		return paymentbroker.PaymentVoucher{}, err
+	}
 
-	wg.Wait()
+	return voucher, nil
 }
 
-func TestPaymentChannelLs(t *testing.T) {
-	t.Parallel()
-	assert := assert.New(t)
+// BasiFastSetup creates a environment and a single node, and environment options
+func BasicFastSetup(t *testing.T, ctx context.Context, fastenvOpts fast.EnvironmentOpts) (fast.Environment, *fast.Filecoin, func() *fast.Filecoin) {
 	require := require.New(t)
 
-	t.Run("Works with default payer", func(t *testing.T) {
-		t.Parallel()
-
-		payer, err := address.NewFromString(fixtures.TestAddresses[2])
-		require.NoError(err)
-		target, err := address.NewFromString(fixtures.TestAddresses[1])
-		require.NoError(err)
-
-		eol := types.NewBlockHeight(20)
-		amt := types.NewAttoFILFromFIL(10000)
-
-		daemonTestWithPaymentChannel(t, &payer, &target, amt, eol, func(d *th.TestDaemon, channelID *types.ChannelID) {
-			ls := listChannelsAsStrs(d, &payer)[0]
-
-			assert.Equal(fmt.Sprintf("%s: target: %s, amt: 10000, amt redeemed: 0, eol: 20", channelID, target.String()), ls)
-		})
-	})
+	// Create a directory for the test using the test name (mostly for FAST)
+	// TODO(tperson) use a different TempDir that uses MkdirAll
+	dir, err := ioutil.TempDir("", strings.Replace(t.Name(), "/", ".", -1))
```

Comment:
`TempDir` only expects that there is a single directory level used in the provided name, and that it's not a path. However, nested tests are separated by `/`, which makes this fail.